src/main/scala/cognite/spark/v1/FlexibleDataModelBaseRelation.scala

```diff
@@ -127,113 +131,67 @@ abstract class FlexibleDataModelBaseRelation(config: RelationConfig, sqlContext:
-          .traverse(
-            toInstanceFilter(instanceType, _, space = space, versionedExternalId = versionedExternalId))
-          .map(FilterDefinition.And.apply)
-      case Or(f1, f2) =>
-        Vector(f1, f2)
-          .traverse(
-            toInstanceFilter(instanceType, _, space = space, versionedExternalId = versionedExternalId))
-          .map(FilterDefinition.Or.apply)
-      //Node types are optional so exists will be automatically checked
-      //Types are a property of the node/edge and should not use the view but directly check on the node
-      case IsNotNull(attribute)
-          if attribute.equalsIgnoreCase("_type") && instanceType == InstanceType.Node =>
-        Right(FilterDefinition.Exists(Seq("node", "type")))
-      case IsNotNull(attribute)
-          if attribute.equalsIgnoreCase("type") && instanceType == InstanceType.Edge =>
-        Right(FilterDefinition.Exists(Seq("edge", "type")))
       case IsNotNull(attribute) =>
         Right(FilterDefinition.Exists(Seq(space, versionedExternalId, attribute)))
-      case IsNull(attribute)
-          if attribute.equalsIgnoreCase("_type") && instanceType == InstanceType.Node =>
-        Right(FilterDefinition.Not(FilterDefinition.Exists(Seq("node", "type"))))
-      case IsNull(attribute)
-          if attribute.equalsIgnoreCase("type") && instanceType == InstanceType.Edge =>
-        Right(FilterDefinition.Not(FilterDefinition.Exists(Seq("edge", "type"))))
       case IsNull(attribute) =>
         Right(FilterDefinition.Not(FilterDefinition.Exists(Seq(space, versionedExternalId, attribute))))
-      case Not(f) =>
-        toInstanceFilter(instanceType, f, space = space, versionedExternalId = versionedExternalId)
-          .map(FilterDefinition.Not.apply)
       case f =>
         Left(
           new CdfSparkIllegalArgumentException(
             s"Unsupported filter '${f.getClass.getSimpleName}', ${f.toString}"))
     }
-  }
 
-  protected def toNodeOrEdgeAttributeFilter(
+  // Some reserved attributes are not attached to a view but directly to the node/edge
+  // This handles filtering on these attributes.
+  private def toNodeOrEdgeAttributeFilter(
       instanceType: InstanceType,
-      sparkFilter: Filter): Either[CdfSparkException, FilterDefinition] =
+      sparkFilter: Filter): Either[CdfSparkException, FilterDefinition] = {
+    val nodeOrEdgeStringAttributes = Seq("space", "externalId")
+    val nodeOrEdgeReferenceAttributes = Seq("type", "_type", "startNode", "endNode")
     sparkFilter match {
-      case EqualTo(attribute, value) if attribute.equalsIgnoreCase("space") =>
+      case EqualTo(attribute, value) if nodeOrEdgeStringAttributes.contains(attribute) =>
         Right(
           FilterDefinition.Equals(
-            createNodeOrEdgeCommonAttributeRef(instanceType, "space"),
-            FilterValueDefinition.String(String.valueOf(value))))
-      case EqualTo(attribute, value) if attribute.equalsIgnoreCase("externalId") =>
```

Comment:
there was ignoreCase part, iirc scala has CIString
not sure if easy to check for real usage with wrong casing and then drop ignorecase

Reply:
Ah yeah might be worth ignoring case everywhere

Reply:
though as you say, it would be weird for a customer to write externalId or other keywords the "wrong" way...  since attributes aren't case insensitive I think this is overeager use of ignore case.

Reply:
It seems like nothing supports case insensitiveness (not querying directly, nor using gaphql) so we are alone in doing this, which raises the question of whether we should.

Reply:
I mean it's more of a hassle than just supporting it, but we aren't testing it or anything, which feels like an undocumented feature and could cause problems down the line

Reply:
Yeah, most likely it's wrong to have it but safer to keep and remove in a separate PR in case of reverts or failing tests

Reply:
Actually it doesn't matter, we have in jetfire set the property:     .config("spark.sql.caseSensitive", "true")
So even though the pushdown filters are case-insensitive, the second pass done in SQL will not be, so it won't work if you use the wrong case right now.